ducktape/tests/reporter.py

```diff
@@ -63,7 +63,7 @@ def result_string(self):
             result_lines.append("    " + self.result.summary)
 
         if self.result.data is not None:
-            result_lines.append(json.dumps(self.result.data))
+            result_lines.append(pprint.pformat(self.result.data))
```

Comment:
These changes seem unrelated to the rest of the patch. Why change to pprint? In the parts below where it writes to `.json` files, is the output guaranteed to be JSON compatible? It seems like the places where we cared about pretty printing we were already properly indenting the JSON.